cmd/proxy/actions/basicauth_test.go

```diff
@@ -0,0 +1,57 @@
+package actions
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+type checkAuthTest struct {
+	name     string
+	user     string
+	pass     string
+	httpUser string
+	httpPass string
+	want     bool
+}
+
+var checkAuthTests = [...]checkAuthTest{
+	{
+		name:     "success",
+		user:     "athens",
+		pass:     "1234",
+		httpUser: "athens",
+		httpPass: "1234",
+		want:     true,
+	},
+	{
+		name:     "username incorrect",
+		user:     "athens",
+		pass:     "1234",
+		httpUser: "not athens",
+		httpPass: "1234",
+		want:     false,
+	},
+	{
+		name:     "password incorrect",
+		user:     "athens",
+		pass:     "1234",
+		httpUser: "athens",
+		httpPass: "not 1234",
+		want:     false,
+	},
+}
+
+func TestCheckAuth(t *testing.T) {
```

Comment:
I like authentication tests regardless of how simple they are. I would like another maintainer to have a look at it though.